src/Ui.GraphiQL/Internal/graphiql.cshtml

```diff
@@ -96,25 +96,34 @@
 		function onEditOperationName(newOperationName) {
 			parameters.operationName = newOperationName;
 			updateURL();
-		}
+        }
 
-		function updateURL() {
-			var newSearch = '?' + Object.keys(parameters).filter(function (key) {
-				return Boolean(parameters[key]);
-			}).map(function (key) {
-				return encodeURIComponent(key) + '=' +
-					encodeURIComponent(parameters[key]);
-			}).join('&');
-			history.replaceState(null, null, newSearch);
-		}
+        function updateURL() {
+            var newSearch = '?' + Object.keys(parameters).filter(function (key) {
+                return Boolean(parameters[key]);
+            }).map(function (key) {
+                return encodeURIComponent(key) + '=' +
+                    encodeURIComponent(parameters[key]);
+            }).join('&');
+            history.replaceState(null, null, newSearch);
+        }
+
+        function stripGraphQLUiEndPointFromPathName(graphQlUiEndpoint) {
+            var currentPathName = window.location.pathname;
+            var regex = new RegExp(graphQlUiEndpoint + "\/*");
+            currentPathName = currentPathName.replace(regex, '');
+            return currentPathName;
+        }
+
+        var strippedCurrentPathName = stripGraphQLUiEndPointFromPathName("@Model.GraphiQLPath");
```

Comment:
This injected variable appears to be encoded for HTML instead of for Javascript - correct?  Or does Razor detect that it's within a `<script>` tag and encode it differently?

Reply:
Razor? No Razor here 🤔 .

Reply:
cshtml are razor pages and `@Model.GraphiQLPath` is razor injecting a variable. By default it will format variables with html encoding. So I always use JavaScript encoding as shown below when injecting a variable into a script tag like this. Or encode it as json and then inject it. Or put it into a html tag and read the tag.

Reply:
> cshtml are razor pages

Yes but ... no Razor engine/stuff/handlers/compilers here.

Reply:
Just good old `string.Replace`.

Reply:
Ok I see.  Then we should modify the `string.Replace` encode to javascript-encode the variables for use on the page.  Probably not a critical fix as the url is controlled by the setup code, not user input.

Reply:
👍